utils/sample_qc.py

```diff
@@ -0,0 +1,487 @@
+        min_male_y_sites_called: int,
+        max_y_female_call_rate: float = 0.15,
+        min_y_male_call_rate: float = 0.8
+) -> hl.Table:
+    """
+    Imputes sample sex based on X-chromosome heterozygosity and Y-callrate (if Y calls are present)
+
+    :param x_mt:
+    :param y_mt:
+    :param platform_ht:
+    :param male_min_f_stat:
+    :param female_max_f_stat:
+    :param min_male_y_sites_called:
+    :param max_y_female_call_rate:
+    :param min_y_male_call_rate:
+    :return:
+    """
+    logger.info("Imputing samples sex")
+
+    x = hl.filter_intervals(x_mt, [hl.parse_locus_interval('X')])
+    x_ht = hl.impute_sex(x.GT, aaf_threshold=0.05, female_threshold=female_max_f_stat, male_threshold=male_min_f_stat)
+    y = hl.filter_intervals(y_mt, [hl.parse_locus_interval('Y')])
+    y = y.filter_rows(y.locus.in_y_nonpar())
+    sex_ht = y.annotate_cols(
+        qc_platform=platform_ht[y.col_key].qc_platform,
+        is_female=x_ht[y.col_key].is_female,
+        y_call_rate=hl.agg.fraction(hl.is_defined(y.GT)),
+        n_y_sites_called=hl.agg.count_where(hl.is_defined(y.GT)),
+        **{f'x_{ann}': x_ht[y.col_key][ann] for ann in x_ht.row_value if ann != 'is_female'}
+    ).cols()
+
+    mean_male_y_sites_called = sex_ht.aggregate(hl.agg.filter(~sex_ht.is_female, hl.agg.group_by(sex_ht.qc_platform, hl.agg.mean(sex_ht.n_y_sites_called))))
+    y_call_rate_stats = sex_ht.aggregate(
+        hl.agg.filter(
+            hl.is_defined(sex_ht.is_female),
+            hl.agg.group_by(hl.tuple([sex_ht.qc_platform, sex_ht.is_female]),
+                            hl.agg.stats(sex_ht.y_call_rate)
+                            )
+        )
+    )
+
+    no_y_call_rate_platforms = set()
+    for platform, sites_called in mean_male_y_sites_called.items():
+        if sites_called < min_male_y_sites_called:
+            logger.warn(f"Mean number of sites in males on Y chromosome for platform {platform} is < {min_male_y_sites_called} ({sites_called} sites found). Y call rate filter will NOT be applied for samples on platform {platform}.")
+            no_y_call_rate_platforms.add(platform)
+
+    sex_ht = sex_ht.annotate_globals(y_call_rate_stats=y_call_rate_stats,
+                                     no_y_call_rate_platforms=no_y_call_rate_platforms if no_y_call_rate_platforms else hl.empty_set(hl.tint32))
+    y_female_stats = sex_ht.y_call_rate_stats[(sex_ht.qc_platform, True)]
+    y_male_stats = sex_ht.y_call_rate_stats[(sex_ht.qc_platform, False)]
+    sex_ht = sex_ht.annotate(
+        is_female=(
+            hl.case()
+                .when(sex_ht.no_y_call_rate_platforms.contains(sex_ht.qc_platform), sex_ht.is_female)
+                .when(sex_ht.is_female & ((sex_ht.y_call_rate - y_female_stats.min)/(y_male_stats.max - y_female_stats.min) < max_y_female_call_rate), True)
+                .when(~sex_ht.is_female & ((sex_ht.y_call_rate - y_female_stats.min)/(y_male_stats.max - y_female_stats.min) > min_y_male_call_rate), False)
```

Comment:
> does this also account for when `is_female` is missing?

Yes, `is_female` missing => missing



> @klaricch's code looks at Y coverage instead of Y callrate, and she uses it to check for sex aneuploidies -- would it be helpful to merge and check both metrics for aneuploidies?

Yes, I think we should merge/decide on what makes most sense.